### PR TITLE
Add client hostname option

### PR DESCRIPTION
**Pull request by OrangeFlag**
## Summary

- Propagate os.Hostname() to query `ClientInfo.ClientHostname`.
- Format custom `ClientName` in clickhouse-go style: application product first, then ch-go product.
- Append Go runtime and OS metadata to `ClientName`, for example:
  `application/v1.85.2 clickhouse/ch-go/0.71.0 (lv:go/1.26.4; os:linux)`.

## Motivation

ClickHouse query logs expose both `client_hostname` and `client_name`. Applications need to use these fields for different attribution dimensions:

- `client_hostname`: pod or host identity.
- `client_name`: application/library identity and versions.

This makes it easier to filter by service/version and separately group by pod/host.

## Checklist
- [x] A human-readable description of the changes was provided to include in CHANGELOG

### CODE REVIEW

**Comment by OrangeFlag** [comment]
@kavirajk can you review this PR?

**Comment by OrangeFlag** [comment]
@kavirajk ping

File: client.go [hunk 1]
```diff
@@ -447,15 +450,8 @@ func Connect(ctx context.Context, conn net.Conn, opt Options) (*Client, error) {
...
-	clientName := proto.Name
 	pkg := pkgVersion.Get()
-	if opt.ClientName == "" {
-		if pkg.Name != "" {
-			clientName = fmt.Sprintf("%s (%s)", clientName, pkg.Name)
-		}
-	} else {
-		clientName = fmt.Sprintf("%s %s", clientName, opt.ClientName)
-	}
+	clientName := formatClientName(opt.ClientName, pkg)
```
Comment (kavirajk) [inline client.go:454]: do you think we can add some tests for this format? to avoid any future regression?

File: query.go [hunk 2]
```diff
@@ -102,7 +102,7 @@ func (c *Client) sendQuery(ctx context.Context, q Query) error {
 			InitialQueryID: q.QueryID,
 			InitialAddress: c.conn.LocalAddr().String(),
 			OSUser:         "",
-			ClientHostname: "",
+			ClientHostname: c.hostname,
```
Comment (kavirajk) [inline query.go:105]: Also some tests to make sure we have non-empty hostname would help?

**Review by kavirajk — COMMENTED** [review]
Thanks @OrangeFlag. Left few minor comments.

**Comment by OrangeFlag** [comment]
@kavirajk fixed all review comments

Please take another look
